File: frontend/src/types/employee.ts
import type { User } from "./auth";
import type { Position } from "./position";

// ─── Employee ────────────────────────────────────────────
export type EmployeeStatus = "Active" | "Inactive" | "Terminated";

export interface Employee {
  id: number;
  firstName: string;
  lastname: string;
  nationalId: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;

  salary: number;
  hireDate: string;
  status: EmployeeStatus;
  positionId: number;
  position?: Position | null;
  user?: User | null;
  createdAt: string;
  isDeleted: boolean;
}

export interface EmployeeProfile extends Employee {
  positionName: string;
  departmentName: string;
}

export interface CreateEmployeeRequest {
  firstName: string;
  lastname: string;
  nationalId: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;

  salary: number;
  hireDate?: string;
  status?: number; // 1=Active, 2=Inactive, 3=Terminated
  positionId: number;
  username: string;
  password: string;
  role: number;
}

export interface UpdateEmployeeRequest {
  firstName?: string;
  lastname?: string;
  nationalId?: string;
  email?: string;
  phoneNumber?: string;
  dateOfBirth?: string;

  salary?: number;
  hireDate?: string;
  status?: number;
  positionId?: number;
  role?: number;
}

// ─── Status enum mapping ─────────────────────────────────
export const STATUS_ENUM_MAP: Record<EmployeeStatus, number> = {
  Active: 1,
  Inactive: 2,
  Terminated: 3,
};

export const STATUS_FROM_NUMBER: Record<number, EmployeeStatus> = {
  1: "Active",
  2: "Inactive",
  3: "Terminated",
};

// ─── Shift options ────────────────────────────────────────
export const SHIFT_OPTIONS = [
  { label: "7:00 AM", value: 7 },
  { label: "8:00 AM", value: 8 },
  { label: "9:00 AM", value: 9 },
  { label: "10:00 AM", value: 10 },
];

File: backend/Domain/Models/Employee.cs
namespace backend.Domain.Models
{
    public class Employee : BaseEntity
    {
        private Employee() { }
        public string FirstName { get; private set; }
        public string Lastname { get; private set; }
        public string NationalId { get; private set; }
        public string Email { get; private set; }
        public string PhoneNumber { get; private set; }
        public DateTime DateOfBirth { get; private set; }

        public decimal Salary { get; private set; }
        public DateTime HireDate { get; private set; }
        public EmployeeStatus? Status { get; private set; }
        public int PositionId { get; private set; }
        public int UserId { get; private set; }

        public Position Position { get; private set; } // Navigation property to Position to do this var positionName = employee.Position.Name;
        public User User { get; set; } // Navigation property to User

        public Employee(
            string firstName,
            string lastname,
            string nationalId,
            string email,
            string phoneNumber,
            DateTime dateOfBirth,

            decimal salary,
            int positionId,
            int userId,
            EmployeeStatus? status)
        {
            FirstName = firstName;
            Lastname = lastname;
            NationalId = nationalId;
            Email = email;
            PhoneNumber = phoneNumber;
            DateOfBirth = dateOfBirth;

            Salary = salary;
            PositionId = positionId;
            UserId = userId;

            HireDate = DateTime.UtcNow;
            Status = status ?? EmployeeStatus.Active;
            CreatedAt = DateTime.UtcNow;
        }

        public void Update(
            string? firstName,
            string? lastname,
            string? nationalId,
            string? email,
            string? phoneNumber,
            DateTime? dateOfBirth,

            decimal? salary,
            DateTime? hireDate,
            EmployeeStatus? status,
            int? positionId)
        {
            FirstName = firstName ?? FirstName;
            Lastname = lastname ?? Lastname;
            NationalId = nationalId ?? NationalId;
            Email = email ?? Email;
            PhoneNumber = phoneNumber ?? PhoneNumber;
            DateOfBirth = dateOfBirth ?? DateOfBirth;

            Salary = salary ?? Salary;
            HireDate = hireDate ?? HireDate;
            Status = status ?? Status;
            PositionId = positionId ?? PositionId;
        }
    }
}